### PR TITLE
added security-related functionality

### PR DESCRIPTION
**Pull request by vadymhrnk**

### CODE REVIEW

**Review by Alexander1iubchenko — APPROVED** [review]
Good job

File: src/main/java/com/example/bookstore/repository/RoleRepository.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.example.bookstore.repository;
+
+import com.example.bookstore.model.Role;
+import org.springframework.data.jpa.repository.JpaRepository;
+
```
Comment (bhdnchui) [inline src/main/java/com/example/bookstore/repository/RoleRepository.java:5]: Add `@Repository`

File: src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        - createTable:
+            tableName: users_roles
+            columns:
+              - column:
+                  name: user_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+                    foreignKeyName: user_foreignKey
+                    references: users(id)
```
Comment (bhdnchui) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml:17]: It's better to add `deleteCascade: true` and  `primaryKey: true`

File: src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+                    foreignKeyName: role_foreignKey
+                    references: roles(id)
```
Comment (bhdnchui) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml:26]: Same.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Great job! Just two minor points.

File: src/main/java/com/example/bookstore/repository/RoleRepository.java (hunk 1 above)
Comment (vadymhrnk) [reply src/main/java/com/example/bookstore/repository/RoleRepository.java:5]: [This is what I found](https://stackoverflow.com/questions/44069367/repository-not-necessary-when-implementing-jparepository) about this annotation. It is old but `JpaRepository <T, ID>` interface still have `@NoRepositoryBean` annotation

Comment (bhdnchui) [reply src/main/java/com/example/bookstore/repository/RoleRepository.java:5]: Yeap! That is a great point.

**Review by bhdnchui — APPROVED** [review]
LGTM

**Pull request merged**